Hole when activate file not written

An untimely ^C can create a partially created virtualenv (this simulates that, internally there's at least one report of this happening):

```
$ make virtualenv_run
	./bin/venv-update \
		venv= virtualenv_run/ --python=python3.5 \
		install= -r requirements.txt -r requirements-dev.txt \
		bootstrap-deps= -r requirements-bootstrap.txt \
		pip-command= pymonkey pip-custom-platform -- pip-faster install --upgrade --prune
> virtualenv virtualenv_run/ --python=python3.5
.: 1: Can't open virtualenv_run/bin/activate

Something went wrong! Sending 'virtualenv_run/' back in time, so make knows it's invalid.
make: *** [virtualenv_run] Error 2
$ make virtualenv_run
	./bin/venv-update \
		venv= virtualenv_run/ --python=python3.5 \
		install= -r requirements.txt -r requirements-dev.txt \
		bootstrap-deps= -r requirements-bootstrap.txt \
		pip-command= pymonkey pip-custom-platform -- pip-faster install --upgrade --prune
> virtualenv virtualenv_run/ --python=python3.5
.: 1: Can't open virtualenv_run/bin/activate

Something went wrong! Sending 'virtualenv_run/' back in time, so make knows it's invalid.
make: *** [virtualenv_run] Error 2
```

One can simulate this by:

```
make virtualenv_run
rm virtualenv_run/bin/activate
touch requirements.txt
make virtualenv_run
```

--- a/setup.py
+++ b/setup.py
@@ -72,5 +72,6 @@ def main():
         },
     )  # :pragma:nocover: covered by tox
 
+
 if __name__ == '__main__':
     exit(main())

--- a/tests/functional/validation.py
+++ b/tests/functional/validation.py
@@ -151,6 +151,25 @@ Removing invalidated virtualenv. (system-site-packages changed, to True)
 
 
 @pytest.mark.usefixtures('pypi_server')
+def test_update_invalidated_missing_activate(tmpdir):
+    with tmpdir.as_cwd():
+        enable_coverage()
+        requirements('')
+
+        venv_update()
+        tmpdir.join('venv/bin/activate').remove()
+
+        out, err = venv_update()
+        err = strip_pip_warnings(err)
+        assert err == "sh: 1: .: Can't open venv/bin/activate\n"
+        out = uncolor(out)
+        assert out.startswith('''\
+> virtualenv venv
+Removing invalidated virtualenv. (could not inspect metadata)
+''')
+
+
+@pytest.mark.usefixtures('pypi_server')
 def it_gives_the_same_python_version_as_we_started_with(tmpdir):
     other_python = OtherPython()
     with tmpdir.as_cwd():

--- a/venv_update.py
+++ b/venv_update.py
@@ -57,6 +57,7 @@ from __future__ import unicode_literals
 
 from os.path import exists
 from os.path import join
+from subprocess import CalledProcessError
 
 __version__ = '1.1.2'
 DEFAULT_VIRTUALENV_PATH = 'venv'
@@ -141,7 +142,7 @@ def info(msg):
 
 
 def check_output(cmd):
-    from subprocess import Popen, PIPE, CalledProcessError
+    from subprocess import Popen, PIPE
     process = Popen(cmd, stdout=PIPE)
     output, _ = process.communicate()
     if process.returncode:
@@ -251,7 +252,10 @@ def get_python_version(interpreter):
 
 
 def invalid_virtualenv_reason(venv_path, source_python, destination_python, options):
-    orig_path = get_original_path(venv_path)
+    try:
+        orig_path = get_original_path(venv_path)
+    except CalledProcessError:
+        return 'could not inspect metadata'
     if not samefile(orig_path, venv_path):
         return 'virtualenv moved %s -> %s' % (timid_relpath(orig_path), timid_relpath(venv_path))
     elif has_system_site_packages(destination_python) != options.system_site_packages:
@@ -434,7 +438,6 @@ def pip_faster(venv_path, pip_command, install, bootstrap_deps):
 
 def raise_on_failure(mainfunc):
     """raise if and only if mainfunc fails"""
-    from subprocess import CalledProcessError
     try:
         errors = mainfunc()
         if errors:

